File: app/src/net/reichholf/dreamdroid/helpers/BundleHelper.java
/* © 2010 Stephan Reichholf <stephan at reichholf dot net>
 * 
 * Licensed under the Create-Commons Attribution-Noncommercial-Share Alike 3.0 Unported
 * http://creativecommons.org/licenses/by-nc-sa/3.0/
 */

package net.reichholf.dreamdroid.helpers;

import androidx.annotation.NonNull;

import java.util.ArrayList;

/**
 * @author sre
 *
 */
public class BundleHelper {
	@NonNull
	public static ArrayList<String> toStringArrayList(@NonNull CharSequence[] strings){
		ArrayList<String> list = new ArrayList<>();
		for (CharSequence string : strings) {
			list.add(string.toString());
		}
		return list;
	}
	
	@NonNull
	public static CharSequence[] toCharSequenceArray(@NonNull ArrayList<String> strings){
		CharSequence[] list = new CharSequence[strings.size()];
		for(int i = 0; i < strings.size(); ++i){
			list[i] = strings.get(i);
		}
		return list;
	}
}

File: app/src/net/reichholf/dreamdroid/helpers/BundleHelper.kt
package net.reichholf.dreamdroid.helpers

import android.content.Intent
import android.os.Bundle
import androidx.core.content.IntentCompat
import androidx.core.os.BundleCompat
import java.io.Serializable

object BundleHelper {
    fun toStringArrayList(strings: Array<CharSequence>): ArrayList<String> {
        val list = ArrayList<String>(strings.size)
        for (string in strings) {
            list.add(string.toString())
        }
        return list
    }

    fun toCharSequenceArray(strings: ArrayList<String>): Array<CharSequence> =
        Array(strings.size) { i ->
            strings[i]
        }
}

inline fun <reified T : Serializable> Bundle.getSerializableCompat(key: String): T? =
    BundleCompat.getSerializable(this, key, T::class.java)

inline fun <reified T : Serializable> Intent.getSerializableExtraCompat(key: String): T? =
    IntentCompat.getSerializableExtra(this, key, T::class.java)
